ddtrace/internal/processor/endpoint_call_counter.py

```diff
@@ -0,0 +1,44 @@
+import typing
+
+import attr
+
+from ddtrace.ext import SpanTypes
+from ddtrace.internal import nogevent
+from ddtrace.internal.compat import ensure_str
+from ddtrace.internal.processor import SpanProcessor
+from ddtrace.span import Span
+
+
+EndpointCountsType = typing.Dict[str, int]
+
+
+@attr.s(eq=False)
+class EndpointCallCounterProcessor(SpanProcessor):
+
+    endpoint_counts = attr.ib(init=False, repr=False, type=EndpointCountsType, factory=lambda: {}, eq=False)
+    _endpoint_counts_lock = attr.ib(init=False, repr=False, factory=nogevent.DoubleLock, eq=False)
```

Comment:
My understanding is that `DoubleLock` is needed when we expect both stdlib threads and gevent coroutines to be using the lock. But in the case of span processor we should only be within a gevent execution. Could this be just a gevent lock?

Reply:
That's because we expect the endpoint (and its instrumentation) to be run by a fully gevent-based application, right? If so then this can be just a gevent lock I suppose.

Reply:
I think we do not have any prior knowledge about what kind of app is going to be using this.

Reply:
Switched the lock type